src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -382,15 +382,18 @@ bool Terminal::SetWindowTitle(std::wstring_view title)
 // - true iff we successfully updated the color table entry.
 bool Terminal::SetColorTableEntry(const size_t tableIndex, const COLORREF dwColor)
 {
-    if (tableIndex > _colorTable.size())
+    try

```

Comment:
Note: I am only looking at this because I am invested in the ctrl+backspace functionality, I have no idea what coding standards are going on here.

You should probably nix the exception handling and use `_colorTable.find` like this:
```
auto colorTableItr = _colorTable.find(tableIndex);
if (colorTableItr != _colorTable.end()) {
  *colorTableItr  = dwColor;
  ...
}
else {
  return false;
}
```

Reply:
`_colorTable` is a `std::array` which doesn't have a `find` function.

Reply:
Ah, in that case you could `std::find(_colorTable.begin(), _colorTable.end(), tableIndex)`! (it just does a linear search)

Reply:
Sticking with random access+exception makes the most sense for this function. Calling it with an invalid value is an exceptional event which requires some really arcane input to repro.